component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPAuthenticator.java

```diff
@@ -0,0 +1,2037 @@
+        }
+    }
+
+    /**
+     * This method will return the configured federatedMobileAttributeKey.
+     *
+     * @param context
+     * @param authenticatorName
+     * @return configured federated Mobile attribute key
+     */
+    private String getFederatedMobileAttributeKey(AuthenticationContext context, String authenticatorName) {
+
+        String federatedVoiceAttributeKey = null;
+        Map<String, String> parametersMap;
+        String tenantDomain = context.getTenantDomain();
+        Object propertiesFromLocal = context.getProperty(IdentityHelperConstants.GET_PROPERTY_FROM_REGISTRY);
+        if (propertiesFromLocal != null || tenantDomain.equals(VoiceOTPConstants.SUPER_TENANT)) {
+            parametersMap = FederatedAuthenticatorUtil.getAuthenticatorConfig(authenticatorName);
+            if (parametersMap != null) {
+                federatedVoiceAttributeKey = parametersMap.get
+                        (VoiceOTPConstants.FEDERATED_MOBILE_ATTRIBUTE_KEY);
+            }
+        } else {
+            federatedVoiceAttributeKey = String.valueOf(context.getProperty
+                    (VoiceOTPConstants.FEDERATED_MOBILE_ATTRIBUTE_KEY));
+        }
+        return federatedVoiceAttributeKey;
+    }
+
+    /**
+     * Check with VoiceOTP flow with user existence.
+     *
+     * @param context      the AuthenticationContext
+     * @param request      the HttpServletRequest
+     * @param response     the HttpServletResponse
+     * @param isUserExists check whether user exist or not
+     * @param username     the UserName
+     * @param queryParams  the queryParams
+     * @param tenantDomain the TenantDomain
+     * @param errorPage    the errorPage
+     * @throws AuthenticationFailedException
+     * @throws VoiceOTPException
+     */
+    private void processVoiceOTPFlow(AuthenticationContext context, HttpServletRequest request,
+                                     HttpServletResponse response, boolean isUserExists, String username,
+                                     String queryParams, String tenantDomain, String errorPage)
+            throws AuthenticationFailedException, VoiceOTPException {
+
+        String mobileNumber = null;
+        if (isUserExists) {
+            boolean isVoiceOTPDisabledByUser = VoiceOTPUtils.isVoiceOTPDisableForLocalUser(username, context);
+            if (log.isDebugEnabled()) {
+                log.debug("Has user enabled Voice OTP : " + isVoiceOTPDisabledByUser);
+            }
+            if (isVoiceOTPDisabledByUser) {
+                // that Enable the Voice OTP in user's Profile. Cannot proceed further without Voice OTP authentication.
+                redirectToErrorPage(response, context, queryParams, VoiceOTPConstants.ERROR_VOICEOTP_DISABLE);
+            } else {
+                mobileNumber = getMobileNumber(request, response, context, username, queryParams);
+            }
+        } else if (VoiceOTPUtils.isSendOTPDirectlyToMobile(context)) {
```

Comment:
What is the config `isSendOTPDirectlyToMobile` is used for?
 Is there a config to disable the redirection to mobile number input page and interrupt the flow, if mobile number is not available in the request?

Reply:
The above if check checks the user existance, Does the  `else if (VoiceOTPUtils.isSendOTPDirectlyToMobile(context))` try to send OTP for users by retrieving the mobile number for the request even if the user does not exist ?
The federated user case is handled separately as well. So what flow is handled here ?

Reply:
> Does the else if (VoiceOTPUtils.isSendOTPDirectlyToMobile(context)) try to send OTP for users by retrieving the mobile number for the request even if the user does not exist ?

Yes . This behavior is same for the sms otp authenticator as well.